infra/Pos/Slotting/Impl/Util.hs

```diff
@@ -17,55 +17,79 @@ import           Pos.Core.Types              (EpochIndex, SlotId (..), Timestamp
                                               mkLocalSlotIndex)
 import           Pos.Util.Util               (leftToPanic)
 
+import           Pos.Slotting.Util           (getEpochSlottingDataEmpatically)
 import           Pos.Slotting.MemState.Class (MonadSlotsData (..))
-import           Pos.Slotting.Types          (EpochSlottingData (..), SlottingData (..))
+import           Pos.Slotting.Types          (EpochSlottingData (..))
 
 -- | Approximate current slot using outdated slotting data.
-approxSlotUsingOutdated :: MonadSlotsData m => EpochIndex -> Timestamp -> m SlotId
+approxSlotUsingOutdated
+    :: (MonadSlotsData m)
+    => EpochIndex
+    -> Timestamp
+    -> m SlotId
 approxSlotUsingOutdated penult t = do
```

Comment:
And also instead of passing penult (current) here it makes more sense to pass last (next). Do you agree? I don't insist on fixing it, but seems to be quite trivial.